src/transformers/models/gpt2/tokenization_gpt2_tf.py

```diff
@@ -0,0 +1,72 @@
+import os
+from typing import Dict, List, Union
+
+import tensorflow as tf
+
+from keras_nlp.tokenizers import BytePairTokenizer
+
+from .tokenization_gpt2 import GPT2Tokenizer
+
+
+class TFGPT2Tokenizer(tf.keras.layers.Layer):
+    """
+    This is an in-graph tokenizer for GPT2. It should be initialized similarly to other tokenizers, using the
+    `from_pretrained()` method. It can also be initialized with the `from_tokenizer()` method, which imports settings
+    from an existing standard tokenizer object.
+
+    In-graph tokenizers, unlike other Hugging Face tokenizers, are actually Keras layers and are designed to be run
+    when the model is called, rather than during preprocessing. As a result, they have somewhat more limited options
+    than standard tokenizer classes. They are most useful when you want to create an end-to-end model that goes
+    straight from `tf.string` inputs to outputs.
+
+    Args:
+        vocab (Dict[str, int]): Vocabulary dict for Byte Pair Tokenizer
+        merges (List[str]): Merges list for Byte Pair Tokenizer
+    """
+
+    def __init__(self, vocab: Dict[str, int], merges: List[str]):
+        super().__init__()
+        self.tf_tokenizer = BytePairTokenizer(vocab, merges)
```

Comment:
We may need to set the `sequence_length` argument based on our padding options, no?

Reply:
Done.